File: app/src/main/java/com/juniperphoton/myerlistandroid/widget/CommonTitleBar.java
package com.juniperphoton.myerlistandroid.widget;


import android.app.Activity;
import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.juniperphoton.myerlistandroid.R;
import com.juniperphoton.myerlistandroid.util.ContextUtil;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CommonTitleBar extends FrameLayout {

    @BindView(R.id.back_iv)
    View backView;

    @BindView(R.id.title_tv)
    TextView textView;

    public CommonTitleBar(final Context context, AttributeSet attrs) {
        super(context, attrs);

        LayoutInflater.from(context).inflate(R.layout.common_title_bar, this, true);

        ButterKnife.bind(this);

        TypedArray array = context.obtainStyledAttributes(attrs, R.styleable.CommonTitleBar);
        String title = array.getString(R.styleable.CommonTitleBar_title);
        textView.setText(title);
        array.recycle();

        backView.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                Activity activity = ContextUtil.getActivity(view);
                if (activity != null) {
                    activity.finish();
                }
            }
        });
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/widget/CommonTitleBar.kt
package com.juniperphoton.myerlist.widget

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.juniperphoton.myerlist.R
import com.juniperphoton.myerlist.extension.getActivity

class CommonTitleBar(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs) {
    @JvmField
    @BindView(R.id.back_iv)
    var backView: View? = null

    @JvmField
    @BindView(R.id.title_tv)
    var textView: TextView? = null

    init {
        LayoutInflater.from(context).inflate(R.layout.common_title_bar, this, true)

        ButterKnife.bind(this)

        val array = context.obtainStyledAttributes(attrs, R.styleable.CommonTitleBar)
        val title = array.getString(R.styleable.CommonTitleBar_title)
        textView!!.text = title
        array.recycle()

        backView!!.setOnClickListener { view ->
            val activity = view.getActivity()
            activity?.finish()
        }
    }
}
